### DISCUSSION TITLE
about the compile error of  lacking some classes

### DISCUSSION THREAD

**Discussion by huhucheng** [keycloak/keycloak#10854] [Q&A]
I pulled the newest keycloak source from the github but when I compiled errors ocurred because there was lack of some classes  .I saw the github again and there is no those java files there and I can't find any information about those errors from innernet .
Could anyone please give me some hint to solve this problem?

The compile errors is like below..

/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/client/HotRodClientEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/client/HotRodClientEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/HotRodMapStorageProviderFactory.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/HotRodMapStorageProviderFactory.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/clientscope/HotRodClientScopeEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/clientscope/HotRodClientScopeEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/HotRodMapStorageProviderFactory.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/HotRodMapStorageProviderFactory.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/group/HotRodGroupEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/group/HotRodGroupEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/HotRodMapStorageProviderFactory.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/HotRodMapStorageProviderFactory.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/client/HotRodProtocolMapperEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/client/HotRodProtocolMapperEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/HotRodMapStorageProviderFactory.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/role/HotRodRoleEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/role/HotRodRoleEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/HotRodMapStorageProviderFactory.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/HotRodMapStorageProviderFactory.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/user/HotRodUserConsentEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/user/HotRodUserConsentEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/HotRodMapStorageProviderFactory.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/user/HotRodUserEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/user/HotRodUserCredentialEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/user/HotRodUserCredentialEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/HotRodMapStorageProviderFactory.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/user/HotRodUserEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/user/HotRodUserEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/user/HotRodUserEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/HotRodMapStorageProviderFactory.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/HotRodMapStorageProviderFactory.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/user/HotRodUserFederatedIdentityEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/user/HotRodUserFederatedIdentityEntity.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/HotRodMapStorageProviderFactory.java
/keycloak-model-map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/user/HotRodUserEntity.java
/keycloak-model-map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/client/delegate/JpaClientDelegateProvider.java
/keycloak-model-map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/clientscope/delegate/JpaClientScopeDelegateProvider.java
/keycloak-model-map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/group/delegate/JpaGroupDelegateProvider.java
/keycloak-model-map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/role/delegate/JpaRoleDelegateProvider.java

**Comment by rhyamada** [comment]
From clean maven install I have to compile thoses classes manually.

Error was got
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project keycloak-model-map-hot-rod: Compilation failure: Compilation failure: 
[ERROR] /home/rhyamada/git/keycloak/model/map-hot-rod/target/generated-sources/annotations/org/keycloak/models/map/storage/hotRod/userSession/HotRodUserSessionEntity$___Marshaller_d168fca3cf72558b16a0dbefdcaa90fd247dc0443a44d66ae0f9448ec662478d.java: error reading /home/rhyamada/git/keycloak/model/map-hot-rod/target/generated-sources/annotations/org/keycloak/models/map/storage/hotRod/userSession/HotRodUserSessionEntity$___Marshaller_d168fca3cf72558b16a0dbefdcaa90fd247dc0443a44d66ae0f9448ec662478d.java; /home/rhyamada/git/keycloak/model/map-hot-rod/target/generated-sources/annotations/org/keycloak/models/map/storage/hotRod/userSession/HotRodUserSessionEntity$___Marshaller_d168fca3cf72558b16a0dbefdcaa90fd247dc0443a44d66ae0f9448ec662478d.java
[ERROR] /home/rhyamada/git/keycloak/model/map-hot-rod/target/generated-sources/annotations/org/keycloak/models/map/storage/hotRod/events/HotRodAdminEventEntity$___Marshaller_83831766253bc789fd9f73b8d4f75614569c841359260b4510b3208ef9264e22.java: error reading /home/rhyamada/git/keycloak/model/map-hot-rod/target/generated-sources/annotations/org/keycloak/models/map/storage/hotRod/events/HotRodAdminEventEntity$___Marshaller_83831766253bc789fd9f73b8d4f75614569c841359260b4510b3208ef9264e22.java; /home/rhyamada/git/keycloak/model/map-hot-rod/target/generated-sources/annotations/org/keycloak/models/map/storage/hotRod/events/HotRodAdminEventEntity$___Marshaller_83831766253bc789fd9f73b8d4f75614569c841359260b4510b3208ef9264e22.java
[ERROR] /home/rhyamada/git/keycloak/model/map-hot-rod/target/generated-sources/annotations/org/keycloak/models/map/storage/hotRod/events/HotRodAdminEventEntitySchemaImpl.java: error reading /home/rhyamada/git/keycloak/model/map-hot-rod/target/generated-sources/annotations/org/keycloak/models/map/storage/hotRod/events/HotRodAdminEventEntitySchemaImpl.java; /home/rhyamada/git/keycloak/model/map-hot-rod/target/generated-sources/annotations/org/keycloak/models/map/storage/hotRod/events/HotRodAdminEventEntitySchemaImpl.java
```

Workaround
```
$ cd $KEYCLOAK_SOURCES/model/map-hot-rod/
$ mvn clean install -DskipTests=true
```

After this keycloak sources have been compiled
